### PR TITLE
Feature/add tests with pest

### PR DESCRIPTION
**Pull request by alessandrobelli**
Modified and created the following tests:
 - Tests\Unit\ExampleTest::test_that_true_is_true
 - Tests\Feature\ApiTokenPermissionsTest::test_api_token_permissions_can_be_updated
 - Tests\Feature\AuthenticationTest::test_login_screen_can_be_rendered
 - Tests\Feature\AuthenticationTest::test_users_can_authenticate_using_the_login_screen
 - Tests\Feature\AuthenticationTest::test_users_can_not_authenticate_with_invalid_password
 - Tests\Feature\BrowserSessionsTest::test_other_browser_sessions_can_be_logged_out
 - Tests\Feature\CreateApiTokenTest::test_api_tokens_can_be_created
 - Tests\Feature\CreateTeamTest::test_teams_can_be_created
 - Tests\Feature\DeleteAccountTest::test_user_accounts_can_be_deleted
 - Tests\Feature\DeleteApiTokenTest::test_api_tokens_can_be_deleted
 - Tests\Feature\DeleteTeamTest::test_teams_cant_be_deleted
 - Tests\Feature\DeleteTeamTest::test_empty_teams_can_be_deleted
 - Tests\Feature\DeleteTeamTest::test_personal_teams_cant_be_deleted
 - Tests\Feature\EmailVerificationTest::test_email_verification_screen_can_be_rendered
 - Tests\Feature\EmailVerificationTest::test_email_can_be_verified
 - Tests\Feature\EmailVerificationTest::test_email_can_not_verified_with_invalid_hash
 - Tests\Feature\ExampleTest::test_the_application_returns_a_successful_response
 - Tests\Feature\InviteTeamMemberTest::test_team_members_can_be_invited_to_team
 - Tests\Feature\InviteTeamMemberTest::test_team_member_invitations_can_be_cancelled
 - Tests\Feature\LeaveTeamTest::test_users_can_leave_teams
 - Tests\Feature\LeaveTeamTest::test_team_owners_cant_leave_their_own_team
 - Tests\Feature\PasswordConfirmationTest::test_confirm_password_screen_can_be_rendered
 - Tests\Feature\PasswordConfirmationTest::test_password_can_be_confirmed
 - Tests\Feature\PasswordConfirmationTest::test_password_is_not_confirmed_with_invalid_password
 - Tests\Feature\PasswordResetTest::test_reset_password_link_screen_can_be_rendered
 - Tests\Feature\PasswordResetTest::test_reset_password_link_can_be_requested
 - Tests\Feature\PasswordResetTest::test_reset_password_screen_can_be_rendered
 - Tests\Feature\PasswordResetTest::test_password_can_be_reset_with_valid_token
 - Tests\Feature\ProfileInformationTest::test_profile_information_can_be_updated
 - Tests\Feature\RegistrationTest::test_registration_screen_can_be_rendered
 - Tests\Feature\RegistrationTest::test_new_users_can_register
 - Tests\Feature\RemoveTeamMemberTest::test_team_members_can_be_removed_from_teams
 - Tests\Feature\RemoveTeamMemberTest::test_team_owner_cannot_be_removed_by_team_members
 - Tests\Feature\RemoveTeamMemberTest::test_admin_can_remove_other_team_members
 - Tests\Feature\TwoFactorAuthenticationSettingsTest::test_two_factor_authentication_can_be_enabled
 - Tests\Feature\TwoFactorAuthenticationSettingsTest::test_recovery_codes_can_be_regenerated
 - Tests\Feature\TwoFactorAuthenticationSettingsTest::test_two_factor_authentication_can_be_disabled
 - Tests\Feature\UpdatePasswordTest::test_password_can_be_updated
 - Tests\Feature\UpdatePasswordTest::test_current_password_must_be_correct
 - Tests\Feature\UpdatePasswordTest::test_new_passwords_must_match
 - Tests\Feature\UpdateTeamMemberRoleTest::test_only_team_owner_can_update_team_member_roles
 - Tests\Feature\UpdateTeamNameTest::test_team_names_can_be_updated.

I also added tests passing in the CI, sorry it took some tries because github CI tests requires some particular setup.

This is a starting point to create more later.

### CODE REVIEW

File: docs/BACKEND-TESTING.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    ```
+
+2. **Render Views with Vite**: Start the development server to render views.
+
+    ```shell
+    npm run dev
+    ```
+
+3. **Prepare Testing Environment**: Make sure to have a `.env.testing` file with the correct database settings. This file should include configurations similar to your main `.env` file but with a separate testing database to avoid data loss.
+
```
Comment (jankapunkt) [inline docs/BACKEND-TESTING.md:22]: You could also add here to copy from `.env.example`

```shell
$ cp .env.example .env.testing
```

File: docs/BACKEND-TESTING.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+### Prerequisites
+
+1. **Install Dependencies**: Ensure all dependencies are installed.
+
+    ```shell
+    npm install
+    ```
+
+2. **Render Views with Vite**: Start the development server to render views.
```
Comment (jankapunkt) [inline docs/BACKEND-TESTING.md:15]: Could you explain in a quick sentence, why it's needed to run the client for these tests and how it's different from the e2e tests?

File: .github/workflows/backend_tests.yml [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+      - name: Copy .env file
+        run: cp .env.example .env
+        working-directory: ./web
+
+      - name: Set Environment Variables
+        run: php artisan key:generate
+        working-directory: ./web
+
+      - name: Set up .env file for database
```
Comment (jankapunkt) [inline .github/workflows/backend_tests.yml:48]: In the docs you write the user needs a `.env.testing` but here is none. How does that work for the CI then?

**Review by jankapunkt — DISMISSED** [review]
General approval, although I have some comments. It's up to you to address them or not as I think they're rather optional.

File: docs/BACKEND-TESTING.md (hunk 2 above)
Comment (alessandrobelli) [reply docs/BACKEND-TESTING.md:15]: TLDR; you can check the relationship between database and frontend are consistent, such as permissions or resource ownership.

They're slightly different from e2e tests (as for my knowledge). Here you can test if an unverified user can see the verification page, but you can also imagine the potential: you can check if a user without the ownership of the project can render the project page, or if a user can delete specific resources.

In `EmailVerificationTests.php`:

```php
    public function test_email_verification_screen_can_be_rendered(): void
    {
        if (! Features::enabled(Features::emailVerification())) {
            $this->markTestSkipped('Email verification not enabled.');

            return;
        }

        $user = User::factory()->withPersonalTeam()->unverified()->create();

        $response = $this->actingAs($user)->get('/email/verify');

        $response->assertStatus(200);
    }
```

File: .github/workflows/backend_tests.yml (hunk 3 above)
Comment (alessandrobelli) [reply .github/workflows/backend_tests.yml:48]: `.env.testing` is not necessary, if that's not present tests are going to use the main database, which is the case of the CI

File: docs/BACKEND-TESTING.md (hunk 1 above)
Comment (alessandrobelli) [reply docs/BACKEND-TESTING.md:22]: Good advice, added to docs 👍

**Review by jankapunkt — APPROVED** [review]

**Pull request merged**